include/zephyr/devicetree/mem.h

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file
+ * @brief Memory Devicetree macro public API header file.
+ */
+
+/*
+ * Copyright (c) 2020, Linaro Ltd.
+ * Copyright (c) 2020 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_MEM_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_MEM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <string.h>
+
+/**
+ * @defgroup devicetree-memory Devicetree memory API
+ * @ingroup devicetree
+ * @{
+ */
+
+/* Inidicates that a ram region is nocache. */
+/* TODO Obtained from zephyr,memory-region.yaml. This should be defined in
+ * just one place.
+ */
+#define MEM_NOCACHE_ATTR "RAM_NOCACHE"
+
+/**
+ * @brief Container for MMIO SRAM information specified in devicetree
```

Comment:
Could be other memory regions, sdram, fram, etc, I wouldn't predispose this to mmio sram bindings as the base binding is the base memory-region one

Reply:
Done